File: src/track_head.cpp
#include <opencv2/opencv.hpp>
#include "head_pose_wrapper.h"

int main() {
    initialize_python();

    cv::VideoCapture cap(2);
    if (!cap.isOpened()) {
        std::cerr << "Error opening video stream or file" << std::endl;
        return -1;
    }

    while (true) {
        cv::Mat frame;
        cap >> frame;
        if (frame.empty())
            break;

        cv::Mat processed_frame = process_frame_cpp(frame);

        cv::imshow("Head Pose Estimation", processed_frame);

        if (cv::waitKey(1) == 'q')
            break;
    }

    cap.release();
    cv::destroyAllWindows();

    finalize_python();

    return 0;
}

File: src/track_head.py
from head_pose_module import *

cap = cv2.VideoCapture(2)  # From Camera

while(cap.isOpened()):

    # Take each frame
    ret, img = cap.read()
    if ret:
        img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
        img = cv2.flip(img, 1)
        img_h, img_w, img_c = img.shape
        text = ''
        res = process_frame(img)
        cv2.imshow('Head Pose', res)
        k = cv2.waitKey(1) & 0xFF
        if k == ord("q"):
            break
    else:
        break

cv2.destroyAllWindows()
cap.release()